lib/chef/exceptions.rb

```diff
@@ -152,8 +152,12 @@ class IllegalVersionConstraint < NotImplementedError; end
 
     class MetadataNotValid < StandardError; end
     class MetadataNotFound < StandardError
-      def initialize
-        super "No metadata.rb or metadata.json!"
+      attr_reader :install_path
+      attr_reader :cookbook_name
+      def initialize(install_path, cookbook_name)
+        @install_path = install_path
+        @cookbook_name = cookbook_name
```

Comment:
I don't think we need to convert these to instance variables - just pass in the method args directly probably. Thoughts?